### PR TITLE
Only show notification on success of update

### LINKED ISSUE #1266: Superfluous success notification when creating a dataset with an invalid field
When we input a wrong value in the dataset creation field, at least on the tags field, and the POST request fails we get a success notification which should not be there.

capture d ecran 2017-11-14 14 44 54

Related to #563

### PR DESCRIPTION
**Pull request by taniki**
FIX #1266 

There is still a remaining usability issue because the form does not highlight the faulty tag yet.

### CODE REVIEW

File: js/components/dataset/form.vue [hunk 1]
```diff
@@ -80,11 +80,13 @@ export default {
             const isValid = this.$refs.form.validate();
 
             if (isValid & !this.hideNotifications) {
+                /*
```
Comment (abulte) [inline js/components/dataset/form.vue]: Remove commented part?

File: js/views/dataset-edit.vue [hunk 2]
```diff
@@ -30,12 +30,24 @@ export default {
     },
     methods: {
         save() {
-            const form = this.$refs.form;
+            let form = this.$refs.form;
```
Comment (abulte) [inline js/views/dataset-edit.vue]: I think this should stay a `const`.

File: js/views/dataset-edit.vue [hunk 3]
```diff
@@ -30,12 +30,24 @@ export default {
...
             if (form.validate()) {
                 this.dataset.update(form.serialize(), (response) => {
-                    this.dataset.on_fetched(response);
+                    this.dataset.on_fetched(response);
```
Comment (abulte) [inline js/views/dataset-edit.vue]: Indentation problem? I’m not sure why this line show up in the diff.

File: js/views/dataset-edit.vue [hunk 4]
```diff
@@ -30,12 +30,24 @@ export default {
...
     methods: {
         save() {
-            const form = this.$refs.form;
+            let form = this.$refs.form;
             if (form.validate()) {
                 this.dataset.update(form.serialize(), (response) => {
-                    this.dataset.on_fetched(response);
+                    this.dataset.on_fetched(response); 
+                    
+                    if(!form.hideNotifications){
```
Comment (abulte) [inline js/views/dataset-edit.vue]: Spaces before and after `if (..) {` (this is what I deduce from existing code)

File: js/views/dataset-edit.vue [hunk 5]
```diff
@@ -30,12 +30,24 @@ export default {
...
+                        this.$dispatch('notify', {
+                            autoclose: true,
+                            title: this._('Changes saved'),
+                            details: this._('Your dataset has been updated.')
+                        });
+                    }
+
                     this.$go({name: 'dataset', params: {oid: this.dataset.id}});
-                }, form.on_error);
+                }, (error)=>{
```
Comment (abulte) [inline js/views/dataset-edit.vue]: Spaces around arrow I think (and maybe drop the `()`)

File: js/views/dataset-edit.vue [hunk 6]
```diff
@@ -30,12 +30,24 @@ export default {
...
+                    this.dataset.loading = false;
```
Comment (taniki) [inline js/views/dataset-edit.vue]: Not sure if this is the right place to do it

File: js/views/dataset-edit.vue [hunk 7]
```diff
@@ -34,8 +34,20 @@ export default {
...
+                            autoclose: true,
+                            title: this._('Changes saved'),
+                            details: this._('Your dataset has been updated.')
+                        });
+                    }
+
                     this.$go({name: 'dataset', params: {oid: this.dataset.id}});
-                }, form.on_error);
+                }, error => {
+                    this.dataset.loading = false;
```
Comment (taniki) [inline js/views/dataset-edit.vue]: Not sure this is the right place for this

File: js/views/dataset-edit.vue (hunk 5 above)
Comment (abulte) [reply js/views/dataset-edit.vue]: Sorry about « drop the () », I just realized we usually leave them (see response above).

File: js/views/dataset-edit.vue (hunk 7 above)
Comment (abulte) [reply js/views/dataset-edit.vue]: What do you mean? It hides the loader, right? Looks good here.

**Pull request merged**